tests/unit_test/test_curve.py

```diff
@@ -0,0 +1,122 @@
+import numpy as np
+import unittest
+from beprof.curve import Curve
+
+
+class TestCurveInit(unittest.TestCase):
+    """
+    Testing Curve initialization and .x .y
+    """
+    def test_list_init(self):
+        simple_lists = [[-12, 1], [-1, 7], [0, 3], [3, 17]]
+        c = Curve(simple_lists)
+        assert list(c.x) == [e[0] for e in simple_lists]
+        assert list(c.y) == [e[1] for e in simple_lists]
+
+    def test_numpy_array_init(self):
+        numpy_array = np.array([[-12, 1], [-1, 7], [0, 3], [3, 17]])
+        c = Curve(numpy_array)
+        assert np.array_equal(c, numpy_array)
+
+    def test_numpy_view_init(self):
+        # todo
+        pass
+
+    def test_empty_init(self):
+        with self.assertRaises(TypeError):
+            Curve()
+        with self.assertRaises(IndexError):
+            Curve([])
+
+    def test_one_point_init(self):
+        c = Curve([[1, 2]])
+        assert c.x == 1
+        assert c.y == 2
+
+    def test_two_point_init(self):
+        array = [[-1, 3], [1, 7]]
+        c = Curve(array)
+        assert np.array_equal(c, array)
+
+
+class TestCurveRescale(unittest.TestCase):
+    def setUp(self):
+        # two the same Curves - one for modification/testing,
+        # one for comparison (unmodified)
+        self.compare_curve = Curve([[0, 0], [5, 5], [10, 10]])
+        # todo: check the x[0] element why it cannot be int
+        self.test_curve = Curve([[0.0, 0], [5, 5], [10, 10]])
+
+    def test_rescale_by_one(self):
+        self.test_curve.rescale(factor=1)
+        assert list(self.compare_curve.x) == list(self.test_curve.x)
+        assert list(self.compare_curve.y) == list(self.test_curve.y)
+
+    def test_rescale_by_negative_one(self):
+        self.test_curve.rescale(factor=-1)
+        assert list(self.compare_curve.x) == list(self.test_curve.x)
+        assert list(self.compare_curve.y) == list(self.test_curve.y * -1)
+
+    def test_rescale_by_zero(self):
+        # post.y should contain some nans and infs
+        self.test_curve.rescale(factor=0)
+        assert list(self.compare_curve.x) == list(self.test_curve.x)
+        assert list(self.compare_curve.y) != list(self.test_curve.y)
+        assert np.isnan(self.test_curve.y[0])
+        assert np.isinf(self.test_curve.y[1])
+
+
+class TestCurveSmooth(unittest.TestCase):
```

Comment:
What about other window lengths (2,3,4,5) ?
Does smooth method works with all window sizes, even and odd ?

Check this one:
http://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.medfilt.html
`Elements of kernel_size should be odd`